addons/point_of_sale/models/stock_move.py

```diff
@@ -0,0 +1,6 @@
+from odoo import api, fields, models
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    pos_order_id = fields.Many2one('pos.order', string="Pos Order")
```

Comment:
No need to add a field on move, on each order you have picking_id and you can take picking_id.move_lines